scripts/utils/train_helper.py

```diff
@@ -27,7 +31,75 @@ def get_selfplay_agent(args, total_training_timesteps, train_types, eval_types,
     )
 
     selfplay_trainer.train_agents(total_train_timesteps=total_training_timesteps)
-    return selfplay_trainer.get_agents()
+    return selfplay_trainer.get_agents()[0], tc
+
+
+def get_selfplay_agent_trained_w_selfplay_types(args,
+                                                pop_total_training_timesteps:int,
+                                                sp_w_sp_total_training_timesteps:int,
+                                                sp_w_sp_train_types:list,
+                                                sp_w_sp_eval_types:list,
+                                                tag:str=None,
+                                                pop_force_training:bool=True,
+                                                sp_w_sp_force_training:bool=True,
+                                                parallel:bool=True) -> tuple:
+    '''
+    Train a SP agent using SP train types. This function will first train a SP agent and then let that
+    agent train with itself and one other unseen teammate (e.g. [SP, SP, SP, SP_H] in a 4-chef layout)
```

Comment:
Is the agent_type (AgentPerformance), SP_H, the same as, AgentPerformance.HIGH, which has a value, 'H'?

Reply:
I believe so. I think that `AgentPerformance` is used to sort teams according to `TeamType` but @avv-va would know better than me

Reply:
SP_H refers to an agent from TeammatesCollection[TeamType.HIGH_FIRST] which should also have a  AgentPerformance.HIGH performance, because AgentPerformance is used to sort the teammates_collection as Joe said.